File: trunk/main/test/com/seovic/coherence/identity/UUIDGeneratorTests.java
package com.seovic.coherence.identity;


import com.seovic.coherence.identity.uuid.UUIDGenerator;

import static org.junit.Assert.*;

import org.junit.Test;


/**
 * Tests for UUIDGenerator.
 * 
 * @author Aleksandar Seovic  2008.11.24
 */
public class UUIDGeneratorTests {

    @Test
    public void testIdGeneration() {
        IdentityGeneratorClient igc = new IdentityGeneratorClient(new UUIDGenerator());

        assertEquals(100, igc.generateIdentities(1, 100).size());
        assertEquals(100, igc.generateIdentities(5, 20).size());
        assertEquals(100, igc.generateIdentities(10, 10).size());
    }
}

File: main.net/test/Coherence.Tools.Tests/Identity/UUIDGeneratorTests.cs
using NUnit.Framework;
using Seovic.Coherence.Identity.UUID;

namespace Seovic.Coherence.Identity
{
    [TestFixture]
    public class UUIDGeneratorTests
    {
        [Test]
        public void TestIdGeneration()
        {
            IdentityGeneratorClient<Tangosol.Util.UUID> igc = new IdentityGeneratorClient<Tangosol.Util.UUID>(
                new UUIDGenerator());

            Assert.AreEqual(100, igc.GenerateIdentities(1, 100).Count);
            Assert.AreEqual(100, igc.GenerateIdentities(5, 20).Count);
            Assert.AreEqual(100, igc.GenerateIdentities(10, 10).Count);
        }
    }
}